opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -1102,6 +1102,21 @@ const mergeEntitiesRaw = async (context, user, targetEntity, sourceEntities, tar
     const completeCategory = getVocabulariesCategories().find(({ key }) => key === targetEntity.category);
     await updateElasticVocabularyValue(sourceEntities.map((s) => s.name), targetEntity.name, completeCategory);
   }
+
+  // Prepare S3 file move
+  // Merge files on S3 and update x_opencti_files path in source => it will be added to target by the merge operation.
+  logApp.info('[OPENCTI] Copying files on S3 before merging x_opencti_files');
+  const sourceEntitiesWithFiles = sourceEntities.filter((entity) => { return entity.x_opencti_files ? entity.x_opencti_files.length > 0 : true; });
+
+  for (let i = 0; i < sourceEntitiesWithFiles.length; i += 1) {
+    if (sourceEntitiesWithFiles[i].x_opencti_files) {
```

Comment:
to ease code reading;
```suggestion
    const entity = sourceEntitiesWithFiles[i];
    if (entity.x_opencti_files) {
```
and use `entity` below